cmd/state/main.go

```diff
@@ -145,13 +146,21 @@ func run(args []string, out output.Outputer) (int, error) {
 	project.RegisterConditional(conditional)
 
 	// Run the actual command
-	cmds := cmdtree.New(primer.New(pj, out, authentication.Get(), prompter, sshell, conditional, args))
+	cmds := cmdtree.New(primer.New(pj, out, authentication.Get(), prompter, sshell, conditional))
 
 	child, err := cmds.Command().Find(args[1:])
 	if err != nil {
 		logging.Debug("Could not find child command, error: %v", err)
 	}
 
+	// Cobra will handle the `--` delimiter if flag parsing is enabled.
+	// If the delimeter is not present we have to disable flag parsing
+	// to ensure flags are passed to the shimmed command rather than
+	// parsed as a flag for `state shim`
+	if !strings.Contains(strings.Join(args, " "), " -- ") {
+		child.SetDisableFlagParsing(true)
+	}
```

Comment:
We can still do this inside cmdtree, just don't pass the arg as part of `prime`. Having it in here does feel dirty.

Reply:
It appears that cobra will use `os.Args` if `cmd.SetArgs` is not called on the command. After this it does a lot of parsing and processing to them. The args I'm using here are before all of the processing so I can inform Cobra on how to process them.